libvast/vast/chunk.hpp

```diff
@@ -78,8 +78,8 @@ class chunk final : public caf::ref_counted {
               std::disjunction<std::is_lvalue_reference<Buffer>,
                                std::is_trivially_move_assignable<Buffer>>>>>
   static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
-    auto view = as_bytes(buffer);
-    return make(view, [buffer = std::move(buffer)]() noexcept {
+    const auto view = as_bytes(buffer);
+    return make(view, [buffer = std::exchange(buffer, {})]() noexcept {
       static_cast<void>(buffer);
```

Comment:
Shouldn't we technically call `buffer.~Buffer()` in the deleter? That seems to be not harder to understand than the `static_cast<void>`, and explicitly deletes the content rather than relying on the fact that the deleter itself gets deleted.

Reply:
We could do this, I don't really mind the change. Feel free to do so in a new PR.